File: Projects/TransportathonHackathon.Clients/angular-ui/src/app/models/response-models/companies/getListDynamicCompanyResponse.ts
import { Employee } from '../../domain-models/employee';

export interface GetListDynamicCompanyResponse {
  appUserId: string;
  email: string;
  userName: string;
  companyName: string;
  driverCount: number;
  completedJobsCount: number;
  employees: Employee[] | null;
}

File: Projects/TransportathonHackathon.API/TransportathonHackathon.Application/Features/Companies/Queries/GetListDynamic/GetListDynamicCompanyResponse.cs
using TransportathonHackathon.Domain.Entities;

namespace TransportathonHackathon.Application.Features.Companies.Queries.GetListDynamic
{
    public class GetListDynamicCompanyResponse
    {
        public Guid AppUserId { get; set; }
        public string Email { get; set; }
        public string UserName { get; set; }
        public string CompanyName { get; set; }
        public int DriverCount { get; set; }
        public int CompletedJobsCount { get; set; }
        public IEnumerable<Employee>? Employees { get; set; }
    }
}
